pkg/sriov/networknodestate.go

```diff
@@ -142,3 +145,62 @@ func (builder *NetworkNodeStateBuilder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WaitUntilSyncStatus waits for the duration of the defined timeout or until the
+// SriovNetworkNodeState gets to a specific syncStatus.
+func (builder *NetworkNodeStateBuilder) WaitUntilSyncStatus(syncStatus string, timeout time.Duration) error {
```

Comment:
I think you need to make sure that Sync is stable for few seconds before close this function. I suspect you can use this function as an example: https://github.com/openshift-kni/eco-goinfra/blob/b2486f52938008d0fa468e4e38a852ec60e9027f/pkg/mco/mcp.go#L248

Reply:
I think it's a little different. This function simply checks that the status is the same. There may be a scenario where the status will change after 5 seconds and this is expected

Reply:
Following our conversation we need to implement multiple functions 
WaitUntilSyncStatus
WaitUntilInSyncStatusFor